docs/Release_Management.md

```diff
@@ -0,0 +1,50 @@
+# Release Management
+
+## Overview
+This document describes **Kubernetes Secrets Store CSI Driver** project release management, which talks about versioning, branching and cadence.
+
+## Legend
+
+- **X.Y.Z** refers to the version (git tag) of Secrets Store CSI Driver that is released. This is the version of the Secrets Store CSI Driver image.
+ 
+- **Milestone** should be designed to include feature sets to accommodate monthly release cycles including test gates. GitHub milestones are used by maintainers to manage each release. PRs and Issues for each release should be created as part of a corresponding milestone.
+
+- **Test gates** should include soak tests and upgrade tests from the last minor version.
+
+
+## Versioning
+This project strictly follows [semantic versioning](https://semver.org/spec/v2.0.0.html). All releases will be of the form _vX.Y.Z_ where X is the major version, Y is the minor version and Z is the patch version.
+
+### Patch releases
+- Patch releases provide users with bug fixes and security fixes. They do not contain new features.
+
+### Minor releases
+- Minor releases contain security and bug fixes as well as _**new features**_.
+
+- They are backwards compatible.
+
+### Major releases
+- Major releases contain breaking changes. Breaking changes refer to schema changes, flag changes, and behavior changes of Secrets Store CSI Driver that may require a clean install during upgrade and it may introduce changes that could break backward compatibility.
+
+- Ideally we will avoid making multiple major releases to be always backward compatible, 
+unless project evolves in important new directions and such release is necessary.
+
+- Secrets Store CSI Driver is currently tracking towards first stable release(v1.0.0) with [this](https://github.com/kubernetes-sigs/secrets-store-csi-driver/milestone/5) milestone.
+
+
+
+## Release Cadence and Branching
+- Secrets Store CSI Driver follows `monthly` release schedule.
+
+- A new release should be created on the _`second Wednesday`_ of each month. This schedule not only allows us to do bug fixes, but also provides an opportunity to address underline image vulnerabilities etc. if any.
+
+- Once the new version is decided(as per above guideline), release candidate branch should be created from `master` with name `vX.Y.Z-rc.W` followed by release with same tag, for eg. if new version is going to be `v0.1.2` then release candidate branch should be `v0.1.2-rc.0`
```

Comment:
there's some discrepancies between this and our current release docs:

https://github.com/kubernetes-sigs/secrets-store-csi-driver/blob/master/docs/RELEASE.md#building-and-releasing

we use `release-<VERSION>` for the branch names (no rc's)

@aramase do you think this or the release docs should be updated? Should we also create long lived release branches?

Reply:
I think we should create long live release branches as part of `v0.1.0`. This could be the 1st release that we can explore options to cherry pick bug fixes and cut patch releases. 

Note: we'll also need to track pipeline changes to handle these scenarios of building from release branches.

As for the release branches, I don't think we need release branches for rc. We cut a release branch for the intended release `release-v0.1.0` and rc images will be published from that release branch with tags. After rc image validation, we'll build a release image from the same branch. WDYT?

*This is how kubernetes does it today*

Reply:
that sgtm, and also the merging to master + cherry picking to release branches